src/control/ZoomPanHandler.ts

```diff
@@ -0,0 +1,480 @@
+    return ratio;
+  }
+
+  /**
+   * Get z-factor
+   * @returns  z-factor
+   */
+  get zFactor(): number {
+    return this._zFactor;
+  }
+
+  /**
+   * Check if x is inverted (right to left is positive) from x bounds
+   * @returns  true if inverted
+   */
+  get isXInverted(): boolean {
+    return this.xBounds[1] < this.xBounds[0];
+  }
+
+  /**
+   * Check if y is inverted (bottom to top is positive) from y bounds
+   * @returns  true if inverted
+   */
+  get isYInverted(): boolean {
+    return this.yBounds[1] < this.yBounds[0];
+  }
+
+  /**
+   * Set z factor
+   * @param  factor
+   */
+  set zFactor(factor: number) {
+    this._zFactor = factor;
+    this.recalculateZoomTransform();
+  }
+
+  /**
+   * Get if enable translate extent (pan limit)
+   * @returns  true if enabled
+   */
+  get enableTranslateExtent(): boolean {
+    return this._enableTranslateExtent;
+  }
+
+  /**
+   * Set enable translate extent (pan limit)
+   * @param  enabled - If should be enabled
+   */
+  set enableTranslateExtent(enabled: boolean) {
+    this._enableTranslateExtent = enabled;
+
+    this.updateTranslateExtent();
+  }
+
+  /**
+   * Update translate extent (pan limits)
+   */
+  updateTranslateExtent(): void {
+    const {
+      width,
+      xSpan,
+      ySpan,
+      zFactor,
+      enableTranslateExtent,
+      translateBoundsX,
+      translateBoundsY
+    } = this;
+
+    let x1: number = -Infinity;
+    let y1: number = -Infinity;
+    let x2: number = +Infinity;
+    let y2: number = +Infinity;
+
+    if (enableTranslateExtent) {
+      const ppu: number = width / xSpan;
+      const h: number = ySpan * ppu * zFactor;
+
+      x1 = translateBoundsX[0] * ppu;
+      x2 = translateBoundsX[1] * ppu;
+      y1 = translateBoundsY[0] * ppu * zFactor;
+      y2 = translateBoundsY[1] * ppu * zFactor;
+    }
+
+    this.zoom.translateExtent([
+      [x1, y1],
+      [x2, y2],
+    ]);
+
+  }
+
+  /**
+   * Create an event object from current state
+   */
+  createEventObject(): OnRescaleEvent {
+    const {
+      scaleX,
+      scaleY,
+      xBounds,
+      yBounds,
+      zFactor,
+      viewportRatio,
+      currentTransform,
+      xRatio,
+      yRatio,
+    } = this;
+
+    return {
+      xScale: scaleX.copy(),
+      yScale: scaleY.copy(),
+      xBounds: xBounds,
+      yBounds: yBounds,
+      zFactor: zFactor,
+      viewportRatio,
+      xRatio: xRatio,
+      yRatio: yRatio,
+      transform: { ...currentTransform },
+    };
+  }
+
+  /**
+   * Update scale
+   */
+  rescale() {
+    const {
+      createEventObject,
+    } = this;
+
+    this.onUpdate(createEventObject());
+  }
+
+  /**
+   * Initialized handler
+   */
+  init(): void{
+    this.zoom = d3
+      .zoom()
+      .scaleExtent([0.1, this.options.maxZoomLevel])
+      .on('zoom', this.onZoom);
+
+      this.container.call(this.zoom);
+  }
+
+  /**
+   * Handle zoom
+   */
+  onZoom(): void{
+    if(!d3.event)return;
```

Comment:
Always brackets { return }

Reply:
I think we need to establish which linting rules to follow.